xaitk_saliency/impls/vis_sal_classifier/occlusion_scoring.py

```diff
@@ -0,0 +1,61 @@
+from xaitk_saliency import ImageClassifierSaliencyMapGenerator
+from xaitk_saliency.utils.masking import weight_regions_by_scalar
+
+import numpy as np
+import warnings
+
+
+class OcclusionScoring (ImageClassifierSaliencyMapGenerator):
+    """
+    This saliency implementation transforms black-box
+    image classification scores into saliency heatmaps. This should
+    require a sequence of per-class confidences predicted on the
+    reference image, a number of per-class confidences as predicted
+    on perturbed images, as well as the masks of the reference image
+    perturbations (as would be output from a
+    `PerturbImage` implementation.
+
+    The perturbation masks used by the following implementation are
+    expected to be of type integer. Masks containing values of type
+    float are rounded to the nearest value and binarized
+    with value 1 replacing values greater than or equal to half of
+    the maximum value in mask after rounding while 0 replaces the rest.
+    """
+
+    def generate(
+            self,
+            image_conf: np.ndarray,
+            perturbed_conf: np.ndarray,
+            perturbed_masks: np.ndarray
+    ) -> np.ndarray:
+
+        if isinstance(perturbed_masks[0][0][0], float):
```

Comment:
This is a good step. I worry this is overly specific however. Am I correct in understanding that this implementation effectively requires boolean masks? If so we could perform the check `if perturbed_masks.dtype != bool:` to enter the below logic.

Reply:
Agreed! it was not independent of shape and could cause issues for integers that are not 0 and 1. Changed it to your suggestion. Thank you.